crates/mun_hir/src/ty/infer.rs

```diff
@@ -400,8 +400,7 @@ impl<'a, D: HirDatabase> InferenceResultBuilder<'a, D> {
                     }
                 }
             }
-            Expr::UnaryOp { .. } => Ty::Unknown,
-            //            Expr::UnaryOp { expr: _, op: _ } => {}
+            Expr::UnaryOp { expr, .. } => self.infer_expr(*expr, &Expectation::none()),
```

Comment:
This would also allow:

```
struct Foo;
fn main() {
  let a = Foo;
  let b = -Foo;
}
```

which doesn't make sense

Reply:
You probably want to check the type of the `infer_expr_inner` call and based on that determine the return type as well as emitting an error for types that cannot be negated or notted. See the [rust analyzer source](https://github.com/rust-analyzer/rust-analyzer/blob/f1a07dbf5559e882f46e79ed2a299cf151b99498/crates/ra_hir_ty/src/infer/expr.rs#L313) for inspiration (we used that a lot ;) )

Reply:
Should the signedness of ints be checked here as well?